File: src/test/java/org/vanautrui/languages/compiler/parsing/astnodes/terminal/FloatConstNodeTest.java
package org.vanautrui.languages.compiler.parsing.astnodes.terminal;

import org.junit.Test;
import org.vanautrui.languages.compiler.lexing.utils.CharacterList;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.lexing.tokens.*;

public class FloatConstNodeTest {

    @Test
    public void test1() throws Exception {
        TokenList list = new TokenList();
        list.add(new OperatorToken("-"));
        list.add(new FloatNonNegativeConstantToken(new CharacterList("4.0")));

        FloatConstNode node = new FloatConstNode(list);
    }

	@Test
    public void test2() throws Exception {
        TokenList list = new TokenList();
        list.add(new FloatNonNegativeConstantToken(new CharacterList("4.0")));

        FloatConstNode node = new FloatConstNode(list);
    }
}

File: dragon-parser/src/test/astnodes/terminal/FloatConstNodeTest.cpp
package org.vanautrui.languages.test.astnodes.terminal;

import org.vanautrui.languages.compiler.lexing.tokens.FloatNonNegativeConstantToken;
import org.vanautrui.languages.compiler.lexing.tokens.OperatorToken;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.parsing.astnodes.terminal.FloatConstNode;

import java.nio.file.Paths;

public final class FloatConstNodeTest {

	public static int test1() {
		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));
			list.add(new OperatorToken("-"));
			list.add(new FloatNonNegativeConstantToken(4.0f));

			final FloatConstNode node = new FloatConstNode(list);
			return 1;
		}catch (Exception e){
			return 0;
		}
	}

	public static int test2() throws Exception {
		try {
			final TokenList list = new TokenList(Paths.get("/dev/null"));
			list.add(new FloatNonNegativeConstantToken(4.0f));

			final FloatConstNode node = new FloatConstNode(list);
			return 1;
		}catch (Exception e){
			return 0;
		}
	}
}
